### PR TITLE
Svc acct

### PR DESCRIPTION
**Pull request by briantkennedy**

### CODE REVIEW

File: cmd/listnamespaces/listnamespaces.go [hunk 1]
```diff
@@ -22,16 +23,39 @@ import (
 
 	"github.com/golang/glog"
 	"github.com/mdruskin/kubernetes-enterprise-control/pkg/client"
+	"github.com/pkg/errors"
 )
 
+var flagSyncNsPoc = flag.Bool("sync_ns_poc", false, "perform POC for syncing namespaces")
```
Comment (filmil) [inline cmd/listnamespaces/listnamespaces.go]: nit: Consider spelling out "poc", at least in the help text.

File: cmd/listnamespaces/listnamespaces.go [hunk 2]
```diff
@@ -22,16 +23,39 @@ import (
...
+
+var flagClientMode = flag.String(
```
Comment (filmil) [inline cmd/listnamespaces/listnamespaces.go]: nit: you can pack all vars in a 'var' section.

If you prefer this, I'm fine with it.

var (

)

File: cmd/listnamespaces/listnamespaces.go [hunk 3]
```diff
@@ -41,26 +65,32 @@ func main() {
 		fmt.Printf(" %s\n", namespace)
 	}
 
+	if *flagSyncNsPoc {
+		syncNsPoc(clusterClient)
+	}
+}
+
+func syncNsPoc(clusterClient *client.Client) {
```
Comment (filmil) [inline cmd/listnamespaces/listnamespaces.go]: Consider renaming the function to something that's more readily understood.

Also,document what it does?

File: examples/service-account/setup-account.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  kubectl create -f ./${sa_cfg}
+else
+  echo "Account ${sa_name} exists, will not create"
+fi
+
+secret_name=$(kubectl get sa/${sa_name} -o=jsonpath='{.secrets[0].name}')
+echo "Account has secret ${secret_name}"
+
+kubectl get secrets/${secret_name} -oyaml > secret.yaml
+echo "Secret in secret.yaml"
```
Comment (filmil) [inline examples/service-account/setup-account.sh]: Perhaps add to .gitignore to prevent oopsies?

File: pkg/client/restconfig/bearertoken.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	caCrtData, err := base64.StdEncoding.DecodeString(caCertB64)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to decode ca crt base64")
+	}
+
+	jsonWebTokenBytes, err := base64.StdEncoding.DecodeString(bearerTokenB64)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to decode token base64")
+	}
+	jsonWebToken := string(jsonWebTokenBytes)
```
Comment (filmil) [inline pkg/client/restconfig/bearertoken.go]: nit: this is the only reference  to this variable,maybe just inline it?

File: pkg/client/restconfig/bearertoken.go [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+package restconfig
+
+import (
+	"encoding/base64"
+
+	"github.com/pkg/errors"
+	"k8s.io/client-go/rest"
+)
+
+func NewBearerTokenConfig(
```
Comment (filmil) [inline pkg/client/restconfig/bearertoken.go]: Care to document the funcition and the parameters?  I quickly get fuzzy on the details of bearer token configuration, so I think there's value to documenting the basic stuff well.

File: cmd/listnamespaces/listnamespaces.go (hunk 2 above)
Comment (briantkennedy) [reply cmd/listnamespaces/listnamespaces.go]: done

File: cmd/listnamespaces/listnamespaces.go (hunk 3 above)
Comment (briantkennedy) [reply cmd/listnamespaces/listnamespaces.go]: done

File: examples/service-account/setup-account.sh (hunk 4 above)
Comment (briantkennedy) [reply examples/service-account/setup-account.sh]: this change adds a .gitignore in examples/service-account

File: pkg/client/restconfig/bearertoken.go (hunk 6 above)
Comment (briantkennedy) [reply pkg/client/restconfig/bearertoken.go]: done

File: pkg/client/restconfig/bearertoken.go (hunk 5 above)
Comment (briantkennedy) [reply pkg/client/restconfig/bearertoken.go]: done

**Comment by briantkennedy** [comment]
comments implemented

**Pull request closed without merging**